--- a/src/core/message/messages/BindProtocol.cpp
+++ b/src/core/message/messages/BindProtocol.cpp
@@ -67,9 +67,12 @@ Hyprwire::CBindProtocolMessage::CBindProtocolMessage(const std::string& protocol
     m_data.push_back(HW_MESSAGE_MAGIC_TYPE_STRING);
     m_data.push_back(0);
     m_data.push_back(0);
-    m_data.append_range(g_messageParser->encodeVarInt(protocol.length()));
-    m_data.append_range(protocol);
+    auto _protLen = g_messageParser->encodeVarInt(protocol.length());
+    m_data.insert(m_data.end(), _protLen.begin(), _protLen.end());
+    m_data.insert(m_data.end(), protocol.begin(), protocol.end());
 
-    m_data.append_range(std::vector<uint8_t>{HW_MESSAGE_MAGIC_TYPE_UINT, 0, 0, 0, 0});
+    auto _magic = std::vector<uint8_t>{HW_MESSAGE_MAGIC_TYPE_UINT, 0, 0, 0, 0};
+    m_data.insert(m_data.end(), _magic.begin(), _magic.end());
